proof/proof_test.go

```diff
@@ -246,3 +248,64 @@ func Example_or2() {
 	// 000000b0  4d 97 a9 bf 1a 28 27 6d  3b 71 04 e1 c0 86 96 08  |M....('m;q......|
 	// Proof verified.
 }
+
+func BenchmarkProof(b *testing.B) {
+	rand := blake2xb.New([]byte("random"))
+	suites := []struct {
+		Suite
+	}{
+		{edwards25519.NewBlakeSHA256Ed25519()},
+		{curve25519.NewBlakeSHA256Curve25519(false)},
+		{curve25519.NewBlakeSHA256Curve25519(true)},
+		{nist.NewBlakeSHA256P256()},
+		{nist.NewBlakeSHA256QR512()},
+	}
+
+	for _, suite := range suites {
+		P := suite.Point().Null()
+
+		sval := map[string]kyber.Scalar{}
+		pval := map[string]kyber.Point{}
+		predicateBuilder := make([]string, 0)
+
+		for i := 0; i < b.N; i++ {
+			s := suite.Scalar().Pick(rand)
+			index := strconv.Itoa(i)
+
+			publicPoint := suite.Point().Mul(s, nil)
+
+			sval["x"+index] = s
+			predicateBuilder = append(predicateBuilder, "x"+index)
+			predicateBuilder = append(predicateBuilder, "B"+index)
+			pval["B"+index] = suite.Point().Base()
+
+			P = suite.Point().Add(P, publicPoint)
+		}
+
+		pval["P"] = P
```

Comment:
this looks like it should be in a sub-benchmark, no?

Reply:
I did not put it in a sub-benchmark because this wasn't really the part I wanted measured, only the time for the prover to build the actual proof and the time the verifier takes, what do you think ?

Reply:
I meant the entire for `i := 0; i < b.N; i++ {` block, not just the `pval` setting.

Why keep it in the "upper function" instead of having it in its own sub-benchmark?